packages/infrastructure-lib/src/mongo_settlement_batch_repo.ts

```diff
@@ -125,6 +131,60 @@ export class MongoSettlementBatchRepo implements ISettlementBatchRepo {
 		}
 	}
 
+	async getBatchesByNameWithPagi(
```

Comment:
This is way too complicated and will be slower than using 2 queries, please use the transfers implementation as reference:
https://github.com/mojaloop/transfers-bc/blob/574b11c036100e0a98fbe2ffe80ecf782a89269d/packages/implementations-lib/src/transfers/mongo_transfers_repo.ts#L124